server/api/routers/datasets/datasets.py

```diff
@@ -252,3 +253,144 @@ async def delete_data(
+        dataset: Dataset name
+        file_hash: Hash of the file to cleanup
+
+    Returns:
+        Cleanup result with deleted chunk count
+    """
+    logger.bind(
+        namespace=namespace,
+        project=project,
+        dataset=dataset,
+        file_hash=file_hash,
+    )
+
+    try:
+        # Get project and dataset config
+        project_obj = ProjectService.get_project(namespace, project)
+        project_dir = ProjectService.get_project_dir(namespace, project)
+
+        dataset_config = next(
+            (ds for ds in (project_obj.config.datasets or []) if ds.name == dataset),
+            None,
+        )
+
+        if not dataset_config:
+            raise HTTPException(
+                status_code=404, detail=f"Dataset '{dataset}' not found"
+            )
+
+        database_name = dataset_config.database
+        if not database_name:
+            raise HTTPException(
+                status_code=400, detail=f"Dataset '{dataset}' has no database configured"
+            )
+
+        # Get database configuration
+        if not project_obj.config.rag or not project_obj.config.rag.databases:
+            raise HTTPException(
+                status_code=400, detail="No databases configured in project"
+            )
+
+        database_config = next(
+            (
+                db
+                for db in project_obj.config.rag.databases
+                if db.name == database_name
+            ),
+            None,
+        )
+
+        if not database_config:
+            raise HTTPException(
+                status_code=404, detail=f"Database '{database_name}' not found"
+            )
+
+        # Initialize vector store
+        import importlib
+        from pathlib import Path
+
+        vector_store_config = database_config.config
+        vector_store_type = (
+            database_config.type.value
+            if hasattr(database_config.type, "value")
+            else str(database_config.type)
+        )
+
+        store_name_lower = vector_store_type.replace("Store", "_store").lower()
+        module_path = f"rag.components.stores.{store_name_lower}"
+
+        try:
+            module = importlib.import_module(module_path)
+            store_class = getattr(module, vector_store_type)
+            vector_store = store_class(
+                config=vector_store_config, project_dir=Path(project_dir)
+            )
+        except (ImportError, AttributeError) as e:
+            raise HTTPException(
+                status_code=500, detail=f"Failed to initialize vector store: {e}"
+            ) from e
+
+        # Initialize document manager and delete chunks
+        from rag.core.document_manager import DocumentManager, DeletionStrategy
+
+        doc_manager = DocumentManager(
+            vector_store=vector_store,
+            config={"enable_soft_delete": False},  # Hard delete for cleanup
+        )
+
+        result = doc_manager.delete_documents(
+            document_hashes=[file_hash],
```

Comment:
I don't think this is doing what you'd think. I believe document hashes are not equivalent to the original file hashes